12.0 improve profiles management

stephansainleger opened the pull request:
Hide the public/position profile useless fields in partner form views.
Hide the public/position profile logic for portal user.

Commit:
[FIX] partner_profiles: add missing add-on dependency

Commit:
[IMP] partner_profiles: filter public profiles in name_search answer

Commit:
[IMP] partner_profiles: delete profiles when main partner deleted

When a main partner is unlinked, all the linked profiles (public
profile and position profiles) are deleted.

Commit:
[IMP] partner_profiles: remove useless fields in public profile view

Hides from the partner public form view the fields and data not
considered as relevant for public profile. The public profile aims to
protect the contact data, but not to replace the other ones which are
considered as "administrative" data.

Commit:
[IMP] partner_profiles_portal: refactoring of user information

Merge the main and public information edition in /my/account portal
page.

Commit:
[IMP] partner_profiles_portal: new portal structures data management

refactors the way a portal user can edit the data of the structures he
is affiliated with.
All main, public and position profiles data are gathered in one
"structure" form.
Adds several navigation improvement (back to structures list,
validation message, ...)

Commit:
[IMP] partner_profiles: remove useless fields in position form view

Commit:
[IMP] partner_profiles_portal: add odoo user in partner form view

Commit:
[IMP] partner_profiles_portal: hide `My structures` button for company users

Commit:
[I18N] partner_profiles: update french translations

Commit:
[I18N] partner_profiles_portal: update french translations

Commit:
[IMP] partner_profiles: prevent company type edit in several cases

Prevents the modification of the company type of res.partner:
- if partner is a public or a position profile
- if the partner has position profiles attached

Also synchronize the company type between main and public profiles.

Commit:
[IMP] partner_profiles: refactoring of position profile creation

Commit:
[IMP] partner_profiles_portal: refactor position profil creation

Commit:
[IMP] partner_profiles: clarify position profiles display

separates the notebook pages of the child_ids and the position
profiles in both company and individual partner form views.

Commit:
[IMP] partner_profiles_portal: add website page edition slots

Commit:
[IMP] partner_profiles_portal: add positions details in portal

Adds several portal views to consult all the positions of a structure,
and their details.

Commit:
[IMP] partner_profiles: add profile in partner kanban view

Commit:
[IMP] partner_profiles: sync active field values between profiles

Change in main profile is synchronized in public and position profiles.
Change in public profile is sy,chronized in main and position profiles.
Change in position profile is NOT synchronized in main and public profiles.

Commit:
[FIX] partner_profiles_portal: fix portal boolean fields handling

clementelabore commented on partner_profiles/models/res_partner.py:
```diff
@@ -117,6 +134,49 @@ def create(self, vals):
+    @api.multi
+    def write(self, vals):
+        super(res_partner, self).write(vals)
+        if (vals.get('active') is True) or (vals.get('active') is False) :
```
tu vérifie si active = True ou si active = False ?? pourquoi ?

stephansainleger replied:
C'est une façon certainement maladroite de juste vérifier la présence d'une clef `active` dans `vals`, pour détecter s'il y a un changement sur ce champs. Mais en relisant, effectivement, il y a surement une manière plus propre de faire ça..

clementelabore replied:
tu peux ecrire : 
`if 'active' in vals:`

Commit:
[I18N] partner_profiles: modify profile name french translations

"Profil principal" -> "Fiche administrative"
"Profil public" -> "Coordonnées publiques"
"Profil position" -> "Fonction"

Commit:
[I18N] partner_profiles_portal: modify profiles french translations

"Profil principal" -> "Fiche administrative"
"Profil public" -> "Coordonnées publiques"
"Profil position" -> "Fonction"

Commit:
[IMP] partner_profiles_portal: improve /my/account name field label

Commit:
[IMP] partner_profiles: remove position profiles from child_ids

Commit:
[IMP] partner_profiles_portal: remove position profile frm child_ids

Commit:
[IMP] partner_profiles_portal: add required asterisk in portal forms

Commit:
[IMP] partner_profiles: apply migration only on contact type partner

Commit:
[I18N] partner_profiles_portal: improve french translations
